lib/matplotlib/axes/_axes.py

```diff
@@ -5772,9 +5772,13 @@ def _interp_grid(X):
                                 "This may lead to incorrectly calculated cell "
                                 "edges, in which case, please supply "
                                 f"explicit cell edges to {funcname}.")
-                        X = np.hstack((X[:, [0]] - dX[:, [0]],
-                                       X[:, :-1] + dX,
-                                       X[:, [-1]] + dX[:, [-1]]))
+                        if isinstance(X, np.ma.core.MaskedArray):
```

Comment:
To be more compact you can use `if np.ma.isMA(X):` or `if np.ma.isMaskedArray(X):`.  Both just call `isinstance(...)`.  If you want to use the explicit `isinstance`, you can still make it more compact be deleting the `.core` part.  The most compact version would be
```
hstack = np.ma.hstack if np.ma.isMA(X) else np.hstack
```

Reply:
:+1: I also cleaned up the handling above this as well.